### ISSUE TITLE
NSURLSession delegate  cycle

### ISSUE THREAD

**Issue by Tliens** [firebase/firebase-ios-sdk#7790]
### [REQUIRED] Step 1: Describe your environment

  * Xcode version: __12.3___
  * Firebase SDK version: _7.8.1____
  * Installation method: `CocoaPods ` (select one)
  * Firebase Component: __GoogleDataTransport___ (Auth, Core, Database, Firestore, Messaging, Storage, etc)

### [REQUIRED] Step 2: Describe the problem
NSURLSession delegate  cycle
`
- (instancetype)initWithTarget:(GDTCORTarget)target
                    conditions:(GDTCORUploadConditions)conditions
                     uploadURL:(NSURL *)uploadURL
                         queue:(dispatch_queue_t)queue
                       storage:(id<GDTCORStoragePromiseProtocol>)storage
              metadataProvider:(id<GDTCCTUploadMetadataProvider>)metadataProvider {
  self = [super init];
  if (self) {
    _uploaderQueue = queue;
    NSURLSessionConfiguration *config = [NSURLSessionConfiguration defaultSessionConfiguration];
    _uploaderSession = [NSURLSession sessionWithConfiguration:config
                                                     delegate:self
                                                delegateQueue:nil];
    _target = target;
    _conditions = conditions;
    _uploadURL = uploadURL;
    _storage = storage;
    _metadataProvider = metadataProvider;
  }
  return self;
}`

urlsession  mabye  `invalidateAndCancel` or `finishTasksAndInvalidate`  ???

please fixed！

**Comment by maksymmalyhin** [comment]
@Tliens Sorry for the trouble. The issue is a duplicate of #7780. The fix is in the main branch of [GoogleDataTransport](https://github.com/google/GoogleDataTransport) repo and is expected to be released soon once testing is complete. You are very welcome to test the fix from the branch if you have a chance. Thank you. 

I'm closing the issue so far, but feel free to reopen if you find out additional details.

**Comment by maksymmalyhin** [comment]
The should be fixed in GoogleDataTransport 8.3.1. Please try to update and let us know if you still have the issue.

**Comment by Tliens** [comment]
thanks
